auth.py

```diff
@@ -0,0 +1,228 @@
+                # should have current password set. Optionally may change password.
+                if current_pass_correct is None:
+                    raise ValueError("Must supply current password to change username")
+                if not current_pass_correct:
+                    raise ValueError("Incorrect current password.")
+
+                self.settings['user'] = new_user
+
+            if new_pass != '':
+                if current_pass_correct is None:
+                    raise ValueError("Must supply current password to change password")
+                if not current_pass_correct:
+                    raise ValueError("Incorrect current password.")
+
+                if new_pass2 != new_pass:  # changing password
+                    raise ValueError("New passwords do not match!")
+
+                self.settings['password'] = new_pass
+
+        else:  # no current password
+            if new_user != '':  # setting username and password
+                if new_pass != '' and new_pass != new_pass2:
+                    raise ValueError("New passwords do not match!")
+                if new_pass == '':
+                    raise ValueError("Must provide password")
+                self.settings['user'] = new_user
+                self.settings['password'] = new_pass
+            else:
+                raise ValueError("Must provide username")
+
+
+class WoTTAuth(BasicAuth):
+    display_name = 'WoTT'
+    name = 'auth_wott'
+    config = {
+        'auth_wott': {
+            'wott_secret_name': 'screenly_credentials',
+        }
+    }
+
+    def __init__(self, settings):
+        super(WoTTAuth, self).__init__(settings)
+        self._fetch_credentials()
+
+    def update_settings(self, current_pass_correct):
+        if not self._fetch_credentials():
+            raise ValueError("Can not read WoTT credential file or login credentials record is incorrect")
+
+    def _fetch_credentials(self):
+        wott_credentials_path = os.path.join(WOTT_CREDENTIALS_PATH, WOTT_SCREENLY_CREDENTIAL_NAME + ".json")
+
+        if 'wott_secret_name' in self.settings and self.settings['wott_secret_name']:
+            screenly_credentials_path = os.path.join(WOTT_CREDENTIALS_PATH, self.settings['wott_secret_name'] + ".json")
+            if os.path.isfile(screenly_credentials_path):
+                wott_credentials_path = screenly_credentials_path
+
+        if os.path.isfile(wott_credentials_path):
+            with open(wott_credentials_path, "r") as credentials_file:
+                credentials = json.load(credentials_file)
+                login_record = credentials.get('login_credentials', '')
+                if not login_record:
+                    return False
+                login_record = login_record.split(':', 1)
+                if len(login_record) == 2:
+                    self.user, password = login_record
+                    self.password = '' if password == '' else hashlib.sha256(password).hexdigest()
```

Comment:
Seems password is stored in raw form in file - I'm not sure if it is fine, but I guess that is wott cred format.

Reply:
Exactly so.

Reply:
It's actually the way that the basic auth is done in Screenly OSE too. It's not good, but that's a problem for another day.